test/suite/languageServerClient.test.ts

```diff
@@ -0,0 +1,107 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import * as vscode from 'vscode';
+import {StripeLanguageClient} from '../../src/languageServerClient';
+
+suite('languageServerClient', () => {
+  let sandbox: sinon.SinonSandbox;
+
+  setup(() => {
+    sandbox = sinon.createSandbox();
+  });
+
+  teardown(() => {
+    sandbox.restore();
+  });
+
+  suite('getDotnetProjectFiles', () => {
+    const workspacePath = '/my/path';
+
+    test('returns empty when no workspaces', async () => {
+      sandbox.stub(vscode.workspace, 'workspaceFolders').value([]);
+      console.log('BAOASPJD');
+      const projectFiles = await StripeLanguageClient.getDotnetProjectFiles();
+      assert.deepStrictEqual(projectFiles, []);
+    });
+
+    test('returns empty when workspaceFolder is undefined', async () => {
+      sandbox.stub(vscode.workspace, 'workspaceFolders').value(undefined);
+      const projectFiles = await StripeLanguageClient.getDotnetProjectFiles();
+      assert.deepStrictEqual(projectFiles, []);
+    });
+
+    test('returns sln project if it exists', async () => {
+      const slnFile = '/project.sln';
+      sandbox.stub(vscode.workspace, 'workspaceFolders').value([
+        {
+          index: 0,
+          name: 'workspace',
+          uri: vscode.Uri.file(workspacePath),
+        },
+      ]);
+      sandbox
+        .stub(vscode.workspace, 'findFiles')
+        .returns(Promise.resolve([vscode.Uri.file(slnFile)]));
+
+      const projectFiles = await StripeLanguageClient.getDotnetProjectFiles();
+      assert.deepStrictEqual(projectFiles, [slnFile]);
+    });
+
+    test('returns csproj if it exists', async () => {
+      const csprojFile = '/project.csproj';
+      sandbox.stub(vscode.workspace, 'workspaceFolders').value([
+        {
+          index: 0,
+          name: 'workspace',
+          uri: vscode.Uri.file(workspacePath),
+        },
+      ]);
+      const fileFilesStub = sandbox.stub(vscode.workspace, 'findFiles');
+      fileFilesStub.onCall(0).returns(Promise.resolve([]));
+      fileFilesStub.onCall(1).returns(Promise.resolve([vscode.Uri.file(csprojFile)]));
```

Comment:
Might be worth commenting what the first and second calls correspond to. Or better, we could make `findFilesStub` return different values depending on the argument.

Reply:
Yeah, I tried the latter at first but I could not get that stub to be invoked when I provided the arguments. Let me give it another try and then I'll just leave some comments if that doesn't work out